homework/src/index.js

```diff
@@ -30,18 +33,84 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  const root = document.getElementById('root');
+  createAndAppend("img", root, { src: "./hyf.png", id: "imgLogo" });
+  const divSelect = createAndAppend("header", root);
+  createAndAppend("p", divSelect, { html: "HYF Repositories : ", id: "pContributors" });
+  const repoSelect = createAndAppend("select", divSelect);
+  const divFlex = createAndAppend("section", root);
+  const divInfo = createAndAppend("article", divFlex, { id: "divInfo" });
+  const divCont = createAndAppend("article", divFlex, { id: "divCont" });
+
+  function main(HYF_REPOS_URL) {
+
+    fetchJSON(HYF_REPOS_URL)
+      .catch(error => {
```

Comment:
The `.catch()` method should (almost) always be placed as the last method in the promise chain. See the link to the fundamental on promises below. When I force an error by tweaking the url a bit I can see that you are rendering the error in the UI but I also see this error message in the console:

```
index.js:56 Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined
    at fetchJSON.catch.then.repositories (index.js:56)
```
After the `catch()` method is executed the JavaScript runtime happily continues executing the next `.then()` in the chain, which gets the value of `undefined` for its `repositories` parameter. That is why you a complaint about trying to use `forEach` of undefined.

https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/promises.md#promise-chaining

Reply:
Corrected